f600233d1b154d507b782f84604bcb12/update/sys_script_include_abc17e0edba4e918b1227ea5f39619fc.xml

```diff
@@ -7,237 +7,568 @@
+                acquisitionPackageRecord.package_status = "TASK_ORDER_AWARDED";
+                acquisitionPackageRecord.update();
+            }
+			return result;
+        }
+        else {
+            throw new Error(JSON.stringify({
+                code: "0503",
+                message: "No CLINs created"
+            }));				
+        }
+	},
+	
+	updatePortfolio: function(portfolioSysId, taskOrderNumber) {
+		const taskOrder = this.getTaskOrderData(taskOrderNumber);
+		const portfolio = new GlideRecord("x_g_dis_atat_portfolio");
+		portfolio.get(portfolioSysId);
+		
+		if (!portfolio) {
+			throw new Error(JSON.stringify({
+				code: "0404",
+				message: "No portfolio found"
+			}));
+		}
+		
+		const taskOrderRecord = new GlideRecord("x_g_dis_atat_task_order");
+        taskOrderRecord.addQuery("task_order_number", taskOrder.taskOrderNumber);
+		taskOrderRecord.query();
+		
+		if (!taskOrderRecord.isValid()){
+			throw new Error(JSON.stringify({
+				code: "0404",
+				message: "No existing task order found"
+			}));
+		} else {
+			taskOrderRecord.next();
+		}
+
+		// Update taskOrder information
+		taskOrderRecord.setValue("pop_start_date", taskOrder.popStartDate);
+        taskOrderRecord.setValue("pop_end_date", taskOrder.popEndDate);
+
+		// Array of new CLIN record sys_ids to associate with the taskOrderRecord
+		let clinSysIds = [];
+		let clin_numbers = [];
+
+		// get all existing CLIN Glide Records related to the taskOrderRecord
+		let existingClinsForTO = this.getCLINGlideRecordsByTO(taskOrderRecord.sys_id);
+		if (existingClinsForTO != null) {
+			[clinSysIds,clin_numbers] = this.resolveCLINData(existingClinsForTO,taskOrder);
+		}
+
+		// Identify new CLINs in the TO details and create them.
+		const newClins = Object.keys(taskOrder.clinData).filter((clin) => !clin_numbers.includes(clin));
+		if (Array.isArray(newClins)) {
+			newClins.forEach((clin) => {
+				const newCLINSysId = this.createCLINGlideRecord(clin, taskOrder.clinData[clin], taskOrderRecord.sys_id);
+				clinSysIds.push(newCLINSysId);
 			});
+		}
 
-			// Find Total Task Order Amount
-			var otherAmounts = header.ProcurementInstrumentAmounts.OtherAmounts;
-			var otherAmountsArray = [];
-			if (!Array.isArray(otherAmounts)) {
-				otherAmountsArray.push(otherAmounts);
-			}
-			else {
-				otherAmountsArray = otherAmounts;
+		taskOrderRecord.setValue("clins", clinSysIds.toString());
+		taskOrderRecord.update();
+	},
+	
+	// Function to loop through existing Glide Records for CLINs associated with the 
+	// given task order and delete records for CLINs that no longer exist in the task 
+	// order and update records for CLINs that do.
+	resolveCLINData: function(clinGlideRecords,taskOrder) {
+		let clinSysIds = [];
+		let existingClins = [];
+		while (clinGlideRecords.next()) {
+			const clin = clinGlideRecords.getValue("clin_number");
+			existingClins.push(clin);
+			if (!Object.keys(taskOrder.clinData).includes(clin)){
+				clinGlideRecords.deleteRecord();
```

Comment:
One thing I forgot to update in the ticket - we should add a new column named `Active` to indicate that it has been soft deleted, and then do an update rather than delete.

Reply:
Do you want it as a boolean? Or a choice of "Active", "Inactive","Deleted", etc?

Reply:
I think `boolean` is fine to just mark a soft delete